refactor(focus-ring): use focus-ring CSS vars for focus states only

dancormier opened the pull request:
[STACKS-532](https://stackoverflow.atlassian.net/browse/STACKS-532)

@giamir I'm curious to get your thoughts on two things:

1) Naming: I went with `translucent` because I figured that was descriptive enough but I'm wondering if you can think of a better name to go with for the added css variables
2) When I update the less snapshots locally, each instance of `:` in selectors gets prefixed with `\\\\`. When the workflow runs on GitHub, it expects "\\". Any idea of why this happens and how to resolve? (This is a minor issue so please don't spend too much time on this!)

dancormier committed:
refactor(focus-ring): use focus-ring CSS vars for focus states only

[STACKS-532](https://stackoverflow.atlassian.net/browse/STACKS-532)

dancormier committed:
Revert tsconfig.json change

dancormier committed:
Update less snapshots

dancormier committed:
Update color.less.test.ts.snap

dancormier committed:
Update color-mixins.less.test.ts.snap

dancormier committed:
Update color.less.test.ts.snap

giamir commented on lib/atomic/misc.less:
```diff
@@ -267,7 +267,7 @@
 //  ----------------------------------------------------------------------------
 .outline-none { outline: 0 !important; }
-.outline-ring { outline: solid var(--su-static4) var(--focus-ring); }
+.outline-ring { outline: solid var(--su-static4) var(--translucent-secondary); }
```
I think the doc for this atomic class will need to be updated:
https://stackoverflow.design/product/base/outline

Furthermore we might want to put a disclaimer that `outline-ring` should not be used to highlight focus states because it is not WCAG compliant. Essentially saying that this atomic class is for  "decorative-only" use and should not convey any semantic meaning.

giamir replied:
If it was not for the breaking changes I would even go as far as suggesting to rename the class to something like `.outline-decorative` or `.outline-decorative-ring`

dancormier replied:
> I think the doc for this atomic class will need to be updated

Currently, it still mimics our focus style (though it references `--translucent-secondary`). We'll probably want it to mimic our focus style after we update to the new style broadly. Considering this, 

> If it was not for the breaking changes I would even go as far as suggesting to rename the class to something like .outline-decorative or .outline-decorative-ring

Ultimately, instead of renaming I'd like to get rid of this atomic class altogether. We reference it in only [a single place](https://github.com/StackEng/StackOverflow/blob/231517c19550c525ed2477285d7884ecea6e5c81/StackOverflow/_Scripts/PostCollections/CreateEditPage.ts#L177) and I think it's a class has very little utility.

**Related**: Every once in a while, I come across some minor breaking change I think would be worth including in the next major release whenever it comes along. Do you think it's reasonable to add Jira issues for these scenarios with a tag like `breaking-change` that way we could find all proposed breaking changes when planning for a major release?

giamir replied:
> Ultimately, instead of renaming I'd like to get rid of this atomic class altogether. We reference it in only [a single place](https://github.com/StackEng/StackOverflow/blob/231517c19550c525ed2477285d7884ecea6e5c81/StackOverflow/_Scripts/PostCollections/CreateEditPage.ts#L177) and I think it's a class has very little utility.

Oh yes, let's remove that. :) 

> Related: Every once in a while, I come across some minor breaking change I think would be worth including in the next major release whenever it comes along. Do you think it's reasonable to add Jira issues for these scenarios with a tag like breaking-change that way we could find all proposed breaking changes when planning for a major release?

That makes sense to me, once we have few of those we could start planning a major release. 
Feel free to go ahead with that. Maybe add also the label `stacks-classic` since in our backlog we have tickets for several Stacks libraries these days.

dancormier replied:
I've created [STACKS-540](https://stackoverflow.atlassian.net/browse/STACKS-540) for removal of `.outline-ring`

giamir reviewed (changes requested):
Overall looks good to me after we take care of updating the atomic class docs. 

> Naming: I went with translucent because I figured that was descriptive enough but I'm wondering if you can think of a better name to go with for the added css variables

I think `translucent` is good given that we don't want to tight the color to a semantic value/meaning at this stage I believe.

> When I update the less snapshots locally, each instance of : in selectors gets prefixed with \\\\. When the workflow runs on GitHub, it expects "\". Any idea of why this happens and how to resolve? (This is a minor issue so please don't spend too much time on this!)

I remember upgrading from vitest 0.x.x to 1.x.x resolved some of those weird backslashes appearing in the snapshots. Have you tried to nuke the node_modules and do a fresh install? In my machine I cannot reproduce that issue (only downgrading vitest would cause the problem). 🤷

dancormier commented:
> I remember upgrading from vitest 0.x.x to 1.x.x resolved some of those weird backslashes appearing in the snapshots. Have you tried to nuke the node_modules and do a fresh install?

Nuking and reinstalling deps fixed the issue 🎉 @giamir thank you!

dancormier committed:
docs(outline): add usage warning; update wording

resolves https://github.com/StackExchange/Stacks/pull/1609#discussion_r1462982561
